client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-item.jsx

```diff
@@ -26,12 +27,56 @@ const EmailForwardingItem = React.createClass( {
 		}
 
 		upgradesActions.deleteEmailForwarding( domain, mailbox, ( error ) => {
-			this.recordEvent( 'deleteClick', domain, mailbox, forward_address, ! Boolean( error ) );
+			this.recordEvent( 'deleteClick', domain, mailbox, forward_address, ! error );
 
 			if ( error ) {
-				notices.error( error.message );
+				notices.error( error.message || this.translate( 'Failed to delete email forwarding record. Please try again or {{contactSupportLink}}contact support{{/contactSupportLink}}.',
+					{
+						components: {
+							contactSupportLink: <a href={ supportPaths.CONTACT }/>
+						}
+					} )
+				);
+			} else {
+				notices.success(
+					this.translate( 'Yay, e-mail forwarding for %(email)s has been successfully deleted.', {
+						args: {
+							email: email
+						}
+					} ), {
+						duration: 5000
+					} );
+			}
+		} );
+	},
+
+	resendVerificationEmail: function() {
+		const { temporary, domain, mailbox, forward_address, email } = this.props.emailData;
+
+		if ( temporary ) {
+			return;
+		}
+
+		upgradesActions.resendVerificationEmailForwarding( domain, mailbox, ( error, response ) => {
+			this.recordEvent( 'resendVerificationClick', domain, mailbox, forward_address, ! error );
+
+			if ( error || ! response.sent ) {
```

Comment:
Is there a chance we're throwing out a valuable error message for a generic one?